protocol-units/settlement/mcr/client/Cargo.toml

```diff
@@ -46,8 +50,6 @@ serde = { workspace = true }
 godfig = { workspace = true }
 
 dot-movement = { workspace = true }
-
-[dev-dependencies]
```

Comment:
Why remove this? We don't want test dependencies to pollute the build graph for anything using the mcr client crate.

Reply:
Ah yes, that's why I don't like tests as binaries. Consider moving them to a separate crate e.g. mcr-settlement-tests, or finding a way to have them as integration tests in the crate `./tests` directory.

Reply:
Or, perhaps, do add an "integration-tests" feature flag guarding these binaries, like we have it elsewhere, and make these dependencies conditional on that.

Reply:
I think we should be able to exclude the testing bin from the release build if we, in the root Cargo.toml:
```rust
[profile.release]
exclude = ["test-mcr-settlement-client"]
```

Reply:
I'm not familiar with the above syntax. I couldn't find it in the docs either. 

I would assume this also wouldn't change the bin size if it does work, just drop that particular binary from the workspace build. 

I couldn't actually find a good way to do this with profiles, so our best option is indeed to stick with a flag and just make sure everyone is using it. Sound good to you?

Reply:
Ah, I meant it would not include the binary in the release build. But fine with the flag.

Reply:
I'm not sure about profiles, it's a weird way to affect the build graph.

I agree that the feature flag is not the best solution, either.
So my alternative suggestions are:
1) a separate test-only crate, or
2) find a way to build and run it as regular Rust tests.

It's less important that the tests require external setup.

Reply:
I think we can move this to a separate PR: https://github.com/movementlabsxyz/movement/discussions/643